### DISCUSSION TITLE
Headless server install

### DISCUSSION THREAD

**Discussion by SenorSmartyPants** [maforget/ComicRackCE#22] [Ideas]
I've had this idea for a while, and now it might be possible.

- UI-less server install to function as a "remote" server that could be connected to from the full GUI CRCE client.

**_Why bother?_** With no UI elements this could be easier to create a version that runs on low power ARM based servers. With it running 24/7 it could also possibly import new comics automatically and sync wirelessly with android clients.

There are many dotnet based projects that run well on low power SBCs.

**Comment by duckpuppy** [comment]
I've also had this idea for a while, and had a couple or runs at it. I've mapped out some of the protocol that CR uses, but it's been a spare time project.

**Reply by Totengeist** [reply]
If you set up a public branch, others could help out.

**Comment by flaser** [comment]
This would require moving CR from the _.NET Framework_ to _.NET Core_, which is a huge task.
Currently big portions of the code still use _.NET Framework.3.5_, so effort is underway to uplift to _.Net Framework 4.8_.

See more here:
https://github.com/maforget/ComicRackCE/discussions/10#discussion-6096227

**Comment by maforget** [comment]
My idea is just to have the initial server settings be done via Command Line Switches. Like say you run it with `-server` and it automatically goes into the tray with the server running. It would be easy to implement without a lot of changes. It would just start a default server with all access. Maybe have some setting be changes from the `ComicRack.ini`.

But is that everyone use case? Just the library sharing?

**Comment by duckpuppy** [comment]
Honestly, I don't use library sharing at all. I'd love a headless server
for syncing to my tablet. I don't use the CR UI outside of importing new
comics, running ComicVine Scraper, Data Manager, and Library Organizer. All
comic reading is done on my tablet. I'd love to be able to sync reading
progress to an always running headless server, but I'd still need the UI
occasionally.

Separate UI and server processes would be ideal for me, but that would mean
switching the DB to a proper locking database so that concurrent access is
safe. SQLite would probably work.

**Reply by maforget** [reply]
In your case what's to stop you from just running the program in the tray icon. You could have it start with windows also. You can do this currently if all you want is to just sync to the tablet. 

As I stated on the front page we need to keep expectations down, we need to keep within the scope of the project, this isn't going to be a rewrite of everything with an external server accessible via a browser with a new UI. 

If simple things can be done I am all for it, but since I get by just leaving it in the tray icon, unless someone explains there use case where this is absolutely necessary, I don't see myself working on it. 

I asked the same in the other thread about Docker, someone explained their usage. They wanted a thin client style with access via the browser. There are already tools like Komga for that. If someone wants to access the ComicRack UI then you have the library sharing, but that requires using a client to access the library. 

If the need isn't for the library sharing, what exactly is the server part in the headless server used for?

**Comment by SenorSmartyPants** [comment]
In my ideal situation I would like a headless docker install that could run on my arm64 SBC for syncing to my tablet. I don't leave a windows PC running 24/7.

**Reply by maforget** [reply]
So in your case it's only for syncing. How would you configure what to sync to the tablet if it's headless, you would need somekind of UI.

Also this means working on Linux, like I said in a lot of place, the more I check the code, I don't believe this would happen. ComicRack uses a lot of Win32 API and windows subsystems.

**Comment by Nadiar** [comment]
After thinking about it more, it would be interesting if the sync/share aspect could run as a service, and then you could start a gui from the same system (or even better if you use another PC to manage it).

**Reply by maforget** [reply]
We would really need to redo the Android Version, because it's a matter of time since it's impossible to use. Right now it requires installing it via ADB, so I don't give the Android Version much time (at least for new devices). So for now I would wait for any change.

**Reply by Nadiar** [reply]
Yeah I was playing around with it the other day, but I have barely any Java knowledge. I managed to decompile it with APKLabs, modify the minimum API to 33, fix the manifest so it would recompile with the MITM patch for the certificate, and it would install on my Pixel 5 test device without bypass enabled... But ~~wireless~~ and storage permissions didn't work. While I was able to give it access to the emulated folder access relatively easily, and drag and drop comic files into it to read, I think people would prefer a top level comics directory.  

And on top of this to make it a viable application you would need to completely remove all of the Twitter integration.

**Reply by maforget** [reply]
it might be a idea, problem is the code is unreadable. Don't know about APKLabs, but all I got was `a.a.a.a`, really hard to figure out what does what.

**Reply by Nadiar** [reply]
You still get it with APKLabs,, the only thing that helped me was wifi component shows up in "WifiSyncService.java" and the library scan turns up similary.  Then I can just toss it into chatGPT and ask it to give me better variable/function names to read it and used its names to figure out what to comment out.
